.github/workflows/finalize-release.yml

```diff
@@ -0,0 +1,60 @@
+name: Finalize release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        type: string
+        description: Version of the release to finalize (e.g. v0.15.0)
+        required: true
+
+env:
+  git-user: github-actions[bot]
+  git-email: 41898282+github-actions[bot]@users.noreply.github.com
+
+jobs:
+  extract-metadata:
+    name: Extract release metadata for ${{ inputs.version }}
+    runs-on: ubuntu-22.04
+    outputs:
+      base-version: ${{ steps.extract.outputs.base-version}}
+    steps:
+      - name: Extract
+        id: extract
+        run: |
+          BASE_VERSION=$(echo "${{ inputs.version }}" | cut -c2- |  awk 'BEGIN{FS=OFS="."}NF--')
+          echo ::set-output name=base-version::$BASE_VERSION
+
+  process-chart:
+    needs: [extract-metadata]
+    uses: ./.github/workflows/process-chart.yml
+    with:
+      release-branch: release-${{ needs.extract-metadata.outputs.base-version }}
+      next-version: ${{ inputs.version }}
+      skip-tagging: true
+    secrets:
+      gh-token: ${{ secrets.GH_TOKEN }}
+
+  finalize-release:
+    needs: process-chart
+    name: Finalize release ${{ inputs.version }}
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: release-${{ needs.extract-metadata.outputs.base-version }}
```

Comment:
it fails here: https://github.com/mszostok/botkube/actions/runs/3135061136/jobs/5090318250

AFAIK you should add `extract-metadata` in `needs`.